radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/model/entities/ValidatorEntity.java

```diff
@@ -111,24 +116,46 @@ public void overwriteDataObject(
       updateValidatorFee(builder, preparedValidatorFee, config);
     } else if (dataObject instanceof ValidatorMetadata metadata) {
       updateMetadata(builder, metadata);
+      updateValidatorSystemMetadataVote(builder);
     } else if (dataObject instanceof ValidatorAllowDelegation allowDelegation) {
       updateAllowDelegation(builder, allowDelegation);
     } else if (dataObject
         instanceof com.radixdlt.api.core.openapitools.model.ValidatorSystemMetadata metadata) {
-      updateValidatorSystemMetadata(builder, metadata);
+      if (metadata.getData().isEmpty()) {
```

Comment:
Could we improve the readability of this code? This long if/else is quite hard to read. Maybe change it to a switch?

Reply:
I meant the whole if/else block not only the new addition. IntelliJ recommends that at line 110.

Reply:
done (changed to swtich with pattern matching)